File: p1/java/app/src/main/java/edu/lehigh/cse262/p1/PrimeDivisors.java
package edu.lehigh.cse262.p1;

import java.util.List;
import java.util.ArrayList;

/** PrimeDivisors is a wrapper class around the function `computeDivisors` */
public class PrimeDivisors {
  /**
   * Compute the prime divisors of `value` and return them as a list
   *
   * @param value The value whose prime divisors are to be computed
   * @return A list of the prime divisors of `value`
   */
  List<Integer> computeDivisors(int value) {
    // [CSE 262] Implement Me!
    ArrayList<Integer> primeFactors = new ArrayList<Integer>(); //list of the divisors, or prime factors, of value
    boolean factorFound = false; //track if a divisor of value is found; if not, value is prime.

    for(int i = 2; i <= value/2; i ++){
        //i is a factor of value:
        if(value%i == 0) { // check to see if int value is evenly divisible by i
            primeFactors.addAll(computeDivisors(i)); //add prime factors of i to the list
            primeFactors.addAll(computeDivisors(value/i)); // add prime factors of value/i to the list
            factorFound = true; //value is not prime
            break; //exit for loop, two factors (i, value/i) have been found
        }
    }
    //base case (value is prime)
    if(!factorFound)
        primeFactors.add(value); //add value to the list of its divisors (value is prime and is its own only divisor)
    return primeFactors;
  }

}

File: p1/python/prime_divisors.py
# prime_divisors: compute the prime factorization of a number

def prime_divisors(n): # expected input is an (integer???)
    n = int(n)
    prime_factors = [] # create an empty list to store the prime factors of n
    found_factors = False # set found_factors to False
    for i in range(2, int(n/2)): # for loop iterating from 2 to n/2
        if n%i == 0: # if i is a factor of n
            prime_factors.append(prime_divisors(i)) # recursively appending prime_divisors(i) to prime_factors
            prime_factors.append(prime_divisors(n/i)) # recursively appending prime_divisors(n/i) to prime_factors
            found_factors = True # set found_factors to True, indicating n is not prime
            break # exit for loop, two factors (i & n/i) have been found
    if not found_factors: # checking to see if the given n is prime
        prime_factors.append(n) # appending the prime n to prime_factors
    return prime_factors # returning the list of prime factors


# if __name__ == "__main__":
#     print(prime_divisors(3214324))